src/flask_pyoidc/provider_configuration.py

```diff
@@ -164,22 +181,45 @@ def ensure_provider_metadata(self):
     def registered_client_metadata(self):
         return self._client_metadata
 
-    def register_client(self, redirect_uris, extra_parameters=None):
+    def register_client(self, client: Client, redirect_uri: str):
+
         if not self._client_metadata:
-            if 'registration_endpoint' not in self._provider_metadata:
+            if not self._provider_metadata['registration_endpoint']:
                 raise ValueError("Can't use dynamic client registration, provider metadata is missing "
                                  "'registration_endpoint'.")
 
             registration_request = self._client_registration_info.to_dict()
-            registration_request['redirect_uris'] = redirect_uris
-            if extra_parameters:
-                registration_request.update(extra_parameters)
-
-            resp = self.requests_session \
-                .post(self._provider_metadata['registration_endpoint'],
-                      json=registration_request,
-                      timeout=self.DEFAULT_REQUEST_TIMEOUT)
-            self._client_metadata = ClientMetadata(redirect_uris=redirect_uris, **resp.json())
+            # Check if the user has passed list of redirect_uris in ClientRegistrationInfo.
+            if 'redirect_uris' not in registration_request:
+                # If not, create it.
+                registration_request['redirect_uris'] = [redirect_uri]
+            # if it's passed, check if it's type list.
+            elif not isinstance(registration_request['redirect_uris'], list):
```

Comment:
I don't think this check really belongs here - would be better in `ClientRegistrationInfo` if needed at all.

Reply:
`ClientRegistrationInfo` is an empty subclass of `OIDCData` so there was nowhere else to keep it.

Reply:
That could be extended to contain this check - I'm thinking in terms of class responsibility it would make more sense to make this check in that class instead of here (where it's also not checked until actually trying to use it).